src/util/cacheService/cacheService.ts

```diff
@@ -106,61 +121,327 @@ export class CacheService {
+        if (!bucketCache) {
+            return undefined;
+        }
+
+
+        const scopeData = bucketCache.scopes.get(scopeName);
+        if (scopeData === undefined) {
+            return undefined;
+        }
+
+        const collectionData = scopeData.collections.get(collectionName);
+        if (collectionData === undefined) {
+            return undefined;
+        }
+
+
+
+        if (!collectionData.timeStamp) {
+            collectionData.timeStamp = currentTimestamp;
+        }
+
+        const collectionTimestamp = new Date(collectionData.timeStamp);
+        let minsDifference = 0
+        const difference = currentTimestamp.getTime() - collectionTimestamp.getTime();
+        if (difference != 0) {
+            minsDifference = difference / (1000 * 60);
+        }
+
+        if (minsDifference > collectionTimeout || minsDifference === 0 || forceRefresh) {
+            if (hasQueryService(connection.services)) {
+                await this.cacheSchemaForCollection(connection, collectionData, result);
+            }
+        }
+
+    }
+
+    // Refreshes the index cache for a single collection
+    private refreshCollectionIndexCache = async (connection: IConnection, bucketName: string, scopeName: string, collectionName: string, indexesResult?: QueryIndex[]) => {
+        if (this.cacheStatus === false) {
+            return undefined;
+        }
+        const bucketCache = this.bucketsData.get(bucketName);
+        if (!bucketCache) {
+            return undefined;
+        }
+
+        const scopeData = bucketCache.scopes.get(scopeName);
+        if (scopeData === undefined) {
+            return undefined;
+        }
+
+        const collectionData = scopeData.collections.get(collectionName);
+        if (collectionData === undefined) {
+            return undefined;
+        }
+        if (hasQueryService(connection.services)) {
+            const indexes = await this.cacheIndexesForCollection(connection, collectionData, indexesResult);
+        }
+    }
+
+    // Updates the Bucket cache if required
+    public updateBucketCache = async (bucketName: string, forceRefresh: boolean) => {
+        const connection = getActiveConnection();
+        if (connection) {
+            const isCacheSuccessful = await this.loadCache(connection);
+            if (isCacheSuccessful) {
+                if (forceRefresh) {
+                    await this.cacheBucket(connection, bucketName);
+                    this.cacheStatus = true;
+                    await this.storeCache(connection);
+                }
+            }
+            else {
+                await this.fullCache(true);
+            }
+        }
+    }
+
+
+    // Updates the Index cache for a particular collection if required
+    public updateCollectionIndexCache = async (connection: IConnection, bucketName: string, scopeName: string, collectionName: string, indexesResult?: QueryIndex[]) => {
+        if (this.cacheStatus === false) {
+            return undefined;
```

Comment:
Shouldn't we do a hard cache refresh if it's not cached? Am I missing something?